### PR TITLE
Unknown variables are now reported in the Error collection

### PR DESCRIPTION
**Pull request by yorek**
When a variable used in the template cannot be resolved, an new VariableResolutionException is added to the Error list so that it is possible to show to the end user which variables are evaluated to null because they do not exists.

### CODE REVIEW

File: src/DotLiquid.Tests/FilterTests.cs [hunk 1]
```diff
@@ -169,7 +169,14 @@ public void TestJoin()
 			Assert.AreEqual("1 2 3 4", new Variable("var | join").Render(_context));
 		}
 
-		[Test]
+        [Test]
```
Comment (haf) [inline src/DotLiquid.Tests/FilterTests.cs]: It would seem this PR mixes indentation.

**Comment by yorek** [comment]
My identation is set to 4...the original source code had it set to 8 maybe?

**Comment by haf** [comment]
Hi @yorek Could you please rebase on master?

**Comment by haf** [comment]
Closing since there was no answer. Can be re-opened.

**Pull request closed without merging**